packages/jupyter-ai/src/components/chat-settings.tsx

```diff
@@ -327,6 +363,87 @@ export function ChatSettings(props: ChatSettingsProps): JSX.Element {
         )}
       </Select>
 
+      {/* Completer language model section */}
+      <h2 className="jp-ai-ChatSettings-header">
+        Completer model
+        {props.completionProvider ? (
+          <UseSignal signal={props.completionProvider.settingsChanged}>
+            {(): JSX.Element => (
+              <CompleterSettingsButton
+                provider={props.completionProvider}
+                openSettings={props.openInlineCompleterSettings}
+                selection={clmProvider}
+              />
+            )}
+          </UseSignal>
+        ) : (
+          <CompleterSettingsButton
+            provider={null}
+            openSettings={props.openInlineCompleterSettings}
+            selection={clmProvider}
+          />
+        )}
+      </h2>
+      <Select
+        value={clmProvider?.registry ? clmProvider.id + ':*' : clmGlobalId}
+        label="Language model"
```

Comment:
Done in 13d0ecf13417c07139f8031e18dca080e2ca0f94.

> UseSignal HOC (which I'd like to avoid if possible, per my previous comment)

Done in 92fa8c6d9d6ffdbc050bcb78a508ce757a4731ed as commented on the previous comment. Can you elaborate on _why_ you would like it to avoid?

Reply:
Done! I've left a reply here: https://github.com/jupyterlab/jupyter-ai/pull/711#discussion_r1589565998